x/chainlink/keeper/keeper_test.go

```diff
@@ -38,55 +38,41 @@ func setupKeeper(t testing.TB) (*Keeper, sdk.Context) {
 	return keeper, ctx
 }
 
-// func setupKeeper(t testing.TB) (*Keeper, sdk.Context) {
-// 	storeKey := sdk.NewKVStoreKey(types.StoreKey)
-// 	memStoreKey := storetypes.NewMemoryStoreKey(types.MemStoreKey)
-
-// 	db := tmdb.NewMemDB()
-// 	stateStore := store.NewCommitMultiStore(db)
-// 	stateStore.MountStoreWithDB(storeKey, sdk.StoreTypeIAVL, db)
-// 	stateStore.MountStoreWithDB(memStoreKey, sdk.StoreTypeMemory, nil)
-// 	require.NoError(t, stateStore.LoadLatestVersion())
-
-// 	registry := codectypes.NewInterfaceRegistry()
-// 	keeper := NewKeeper(codec.NewProtoCodec(registry), storeKey, memStoreKey)
-
-// 	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
-// 	return keeper, ctx
-// }
-
 func TestFeedKeyStructure(t *testing.T) {
 	k, ctx := setupKeeper(t)
 	roundStore := ctx.KVStore(k.roundStoreKey)
 	feedStore := ctx.KVStore(k.feedDataStoreKey)
 
-	var tests = []struct {
-		feedId  string
-		roundId uint64
+	testCases := []struct {
+		feedId   string
+		roundIds []uint64
 	}{
-		{"test1", 1111},
-		{"test11", 111},
-		{"test111", 11},
-		{"test1111", 1},
+		{feedId: "test1", roundIds: []uint64{1, 11, 111, 1111}},
+		{feedId: "test11", roundIds: []uint64{1, 11, 111, 1111}},
+		{feedId: "test111", roundIds: []uint64{1, 11, 111, 1111}},
+		{feedId: "test1111", roundIds: []uint64{1, 11, 111, 1111}},
```

Comment:
still dont see a test case for the malformed `{"test11/1", 111}` case

Reply:
this case will require some changes to the keeper logic

Reply:
My bad, I didn't see your feedId case correctly the first time.
I think still we want to allow module owner to add feedId with `/` in it?
Since key are type `[]byte` maybe we can use binary format for this.

Reply:
there should be some way to check if there is a `/` then. it doesnt matter if its bytes because it will just be string encoded. so we'll need to edit the `keeper` logic so that you can verify/validate this